lib/routes/hanime1/previews.ts

```diff
@@ -0,0 +1,86 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import puppeteer from 'puppeteer';
+
+export const route: Route = {
+    path: '/previews/:date',
+    name: '新番预告',
+    maintainers: ['kjasn'],
+    example: '/hanime1/previews/202504',
+    categories: ['anime'],
+    parameters: { date: { description: 'Date in YYYYMM format' } },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: true,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['hanime1.me/previews/:date'],
+            target: '/previews/:date',
+        },
+    ],
+    handler: async (ctx) => {
+        const { date } = ctx.req.param();
+        const url = `https://hanime1.me/previews/${date}`;
+
+        const browser = await puppeteer.launch({
+            headless: true,
+        });
+        const page = await browser.newPage();
+        await page.setUserAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36');
+        await page.setExtraHTTPHeaders({
+            Referer: 'https://hanime1.me/',
+        });
+        await page.goto(url, { waitUntil: 'domcontentloaded' });
+        const html = await page.content();
+        await browser.close();
```

Comment:
Unnecessary use of puppeteer. This site works when using [RSSHub's UA](https://github.com/DIYgod/RSSHub/blob/8508789087d56576430614f6b81de91fdd1f59a7/lib/config.ts#L458).